model/map-jpa/src/main/resources/META-INF/auth-sessions/jpa-auth-sessions-changelog-1.xml

```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Copyright 2022 Red Hat, Inc. and/or its affiliates
+and other contributors as indicated by the @author tags.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd
+                    http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd">
+
+    <!-- format of id of changeSet: auth-sessions-${org.keycloak.models.map.storage.jpa.Constants.CURRENT_SCHEMA_VERSION_AUTH_SESSION} -->
+    <changeSet author="keycloak" id="auth-sessions-1">
+
+        <createTable tableName="kc_auth_root_session">
+            <column name="id" type="UUID">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="metadata" type="json"/>
+        </createTable>
+        <ext:addGeneratedColumn tableName="kc_auth_root_session">
+            <ext:column name="entityversion" type="INTEGER" jsonColumn="metadata" jsonProperty="entityVersion"/>
+            <ext:column name="realmid" type="VARCHAR(36)" jsonColumn="metadata" jsonProperty="fRealmId"/>
+            <ext:column name="timestamp" type="INTEGER" jsonColumn="metadata" jsonProperty="fTimestamp"/>
+            <ext:column name="expiration" type="BIGINT" jsonColumn="metadata" jsonProperty="fExpiration"/>
```

Comment:
I see that "timestamp" is an integer (as it is also within Keycloak's logical layer), and there is "expiration" (with is AFAIK also a timestamp, but as a long). 

I wonder if we should treat both as long within the map storage (across all modules) to make this consistent. 

And if we should place it into the database as a timestamp, although I know having timestamps in databases has also its issues with regards to timezones.

Reply:
> I wonder if we should treat both as long within the map storage (across all modules) to make this consistent.

Good catch. Yes, we should, could you please introduce a GHI for that?

> I also wonder if we should place it into the database as a timestamp, although I know having timestamps in databases has also its issues with regards to time zones.

I'd rather keep it as `BIGINT`, that corresponds to the usage of it within Keycloak. If a view of it should be needed by an admin, they can use DB functions for conversions in SELECTs

Reply:
New GHI: #10960 - beware of the Epochalypse!

As there is a new issue to handle this, this review comment is thereby resolved.